File: app/src/main/kotlin/co/uk/androidrecruitmenttask/StarWarsApplication.java
package co.uk.androidrecruitmenttask;

import android.app.Application;

import co.uk.androidrecruitmenttask.data.api.ApiComponent;
import co.uk.androidrecruitmenttask.data.api.ApiModule;
import co.uk.androidrecruitmenttask.data.api.DaggerApiComponent;

/**
 * Created by filipradon on 10/04/17.
 */

public class StarWarsApplication extends Application {

    private ApiComponent apiComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        apiComponent = DaggerApiComponent.builder()
                .apiModule(new ApiModule())
                .build();
    }

    public ApiComponent getApiComponent() {
        return apiComponent;
    }
}

File: app/src/main/kotlin/co/uk/androidrecruitmenttask/StarWarsApplication.kt
package co.uk.androidrecruitmenttask

import android.app.Activity
import android.app.Application
import co.uk.androidrecruitmenttask.util.injection.DaggerAppComponent
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasActivityInjector
import javax.inject.Inject

class StarWarsApplication : Application(), HasActivityInjector {

    @Inject
    lateinit var activityInjector: DispatchingAndroidInjector<Activity>

    override fun activityInjector(): AndroidInjector<Activity> =
            activityInjector

    override fun onCreate() {
        super.onCreate()
        init()
    }

    private fun init() {
        initAppComponent()
    }

    private fun initAppComponent() {
        DaggerAppComponent.builder()
                .application(this)
                .build()
                .inject(this)
    }
}
